src/filefingerprint.cpp

```diff
@@ -373,6 +360,31 @@ int FileFingerprint::unserializefingerprint(string* d)
     return 1;
 }
 
+size_t FileFingerprint::getHash() const
+{
+    assert(isvalid);
+    size_t value = 0;
+    hashCombine(value, size);
+    hashCombine(value, mtime);
+    for (const auto val : crc)
+    {
+        hashCombine(value, val);
+    }
+    return value;
+}
+
+FileFingerprint& FileFingerprint::operator=(const FileFingerprint& other)
+{
+    if (this != &other)
```

Comment:
I don't think we need to add this self-assignment check, since the self-assignment case will work ok without it, and the test is just extra work that has to be done every other time (I doubt we have any self assignment of these anyway)

Reply:
I will add an assert instead